packages/swing-store/src/transcriptStore.js

```diff
@@ -667,19 +816,36 @@ export function makeTranscriptStore(
     }
     // 'archival' and 'debug' modes accept all spans
 
+    const doCompression =
+      enableTranscriptRolloverCompression && !metadata.isCurrent;
```

Comment:
I like this pattern better because it parallels what we do when the items are arriving as the result of deliveries.  I'm not sure what benefit providing a distinct flag buys us.

Reply:
@warner and I discussed this and settled on `useTranscriptCompression`.